packages/flutter/test/widgets/scroll_position_test.dart

```diff
@@ -160,6 +229,110 @@ void main() {
     expect(count, 1);
   });
 
+  testWidgets('shrink listview', (WidgetTester tester) async {
+    // widget tests run at (800x600)@3.0x
+    await tester.pumpWidget(MaterialApp(
+      home: ListView.builder(
+        physics: const _JumpingScrollPhysics(),
+        itemBuilder: (BuildContext context, int index) => index == 0
+              ? const ExpandingBox(collapsedSize: 400, expandedsize: 1200)
+              : Container(height: 300, color: Colors.red),
+        itemCount: 2,
+      ),
+    ));
+
+    final ScrollPosition position =
+        tester.state<ScrollableState>(find.byType(Scrollable)).position;
+    expect(position.activity.runtimeType, IdleScrollActivity);
+    expect(position.minScrollExtent, 0.0);
+    expect(position.maxScrollExtent, 100.0);
+    expect(position.pixels, 0.0);
+    await tester.tap(find.byType(FlatButton));
+    await tester.pumpAndSettle();
```

Comment:
you use `pumpAndSettle` a lot in this test; in general we should use `pump`, and `pumpAndSettle` is reserved for cases where we want to count the number of frames that an animation takes.